lib/signaling/v3/room.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+const { createTwilioError } = require('../../util/twilio-video-errors');
+const RoomV2 = require('../v2/room');
+const RemoteParticipantV3 = require('../v3/remoteparticipant');
+const TrackSubscriptionsSignaling = require('./tracksubscriptionssignaling');
+
+/**
+ * @extends RoomV2
+ */
+class RoomV3 extends RoomV2 {
+  constructor(
+    localParticipant,
+    initialState,
+    transport,
+    peerConnectionManager,
+    options
+  ) {
+    options = Object.assign({
+      RemoteParticipantSignaling: RemoteParticipantV3,
+      TrackSubscriptionsSignaling
+    }, options);
+
+    options.otherMediaSignalings = [{
+      MediaSignaling: options.TrackSubscriptionsSignaling,
+      initMethod: '_initTrackSubscriptionsSignaling',
+      name: '_trackSubscriptionsSignaling'
+    }];
+
+    super(
+      localParticipant,
+      initialState,
+      transport,
+      peerConnectionManager,
+      options
+    );
+
+    Object.defineProperties(this, {
+      _pendingTrackMids: {
+        value: new Map()
+      }
+    });
+  }
+
+  /**
+   * @private
+   * @override
+   */
+  _addTrackReceiver(trackReceiver) {
+    const idType = trackReceiver.kind === 'data' ? 'id' : 'mid';
+    const deferred = this._getOrCreateTrackReceiverDeferred(trackReceiver[idType], idType);
+    deferred.resolve(trackReceiver);
+    return this;
+  }
+
+  /**
+   * @private
+   * @override
+   */
+  _createRemoteParticipant(participantState) {
+    const { _RemoteParticipantSignaling: RemoteParticipantV3 } = this;
+    return new RemoteParticipantV3(
+      participantState,
+      trackSid => this._getPendingTrackReceiver(trackSid),
+      trackSid => this._getInitialTrackSwitchOffState(trackSid),
+      (trackSid, priority) => this._trackPrioritySignaling.sendTrackPriorityUpdate(trackSid, 'subscribe', priority),
+      (trackSid, hint) => this._renderHintsSignaling.setTrackHint(trackSid, hint),
+      trackSid => this._renderHintsSignaling.clearTrackHint(trackSid)
+    );
```

Comment:
this parameter list is getting too long, and making this caller code to hard to read. How about we pass these params as an object. This will make it more readable for unit tests as well.
```suggestion
    return new RemoteParticipantV3({
      participantState: participantState,
      getReceiver: trackSid => this._getPendingTrackReceiver(trackSid),
      getSwitchOfState: trackSid => this._getInitialTrackSwitchOffState(trackSid),
      setPriority: (trackSid, priority) => this._trackPrioritySignaling.sendTrackPriorityUpdate(trackSid, 'subscribe', priority),
      setRenderHint: (trackSid, hint) => this._renderHintsSignaling.setTrackHint(trackSid, hint),
      clearRenderHint: trackSid => this._renderHintsSignaling.clearTrackHint(trackSid)
    });
```

Reply:
See my previous [response](https://github.com/twilio/twilio-video.js/pull/1732/files#r828481248).